docs/proposals/PROPOSAL_B.md

```diff
@@ -0,0 +1,534 @@
+  "config": {
+    "mediaType": "application/vnd.oci.empty.v1+json",
+    "digest": "sha256:44136fa355b3678a1146ad16f7e8649e94fb4fc21fe77e8310c060f61caaff8a",
+    "size": 2
+  },
+  "layers": [
+    {
+      "mediaType": "application/vnd.oci.image-compatibility.spec.v1+json",
+      "digest": "sha256:4a47f8ae4c713906618413cb9795824d09eeadf948729e213a1ba11a1e31d052",
+      "size": 1710
+    }
+  ],
+  "annotations": {
+    "oci.opencontainers.image.created": "2024-01-02T03:04:05Z"
+  }
+}
+```
+
+### Relation of the Artifact to Image Manifest
+
+```text
+ Image Index                                                 
+ +----------------------------------------------------------+
+ |                                                          |
+ |  Image Manifest A             Image Manifest B           |
+ |  +-----------------------+    +-----------------------+  |
+ |  |platform:              |    |platform:              |  |
+ |  |  archictecture: amd64 |    |  archictecture: arm   |  |
+ |  |  os: linux            |    |  os: linux            |  |
+ |  |                       |    |                       |  |
+ |  |digest: abc123         |    |digest: def456         |  |
+ |  +--------------------|--+    +--------------------|--+  |
+ |                       |                            |     |
+ +-----------------------|----------------------------|-----+
+                         |                            |      
+                         |                            |      
+    Artifact A           |       Artifact B           |      
+    +--------------------|--+    +--------------------|--+   
+    |                    |  |    |                    |  |   
+    | Subject Descriptor |  |    | Subject Descriptor |  |   
+    | +------------------|-+|    | +------------------|-+|   
+    | | digest: abc123     ||    | | digest: def456     ||   
+    | +--------------------+|    | +--------------------+|   
+    |                       |    |                       |   
+    +-----------------------+    +-----------------------+   
+```
+
+The design imposes a strong 1 to 1 relationship between the compatibility artifact and the image.
+That allows users to independently release artifacts and attach compatibility to the already existing images.
+
+The disadvantage of this solution is that you cannot create compatibility that points to multiple images having the same requirements.
+
+### Artifact Discovery
+
+If a registry supports referrers API, it should be used for the artifact discovery.
+
+Otherwise, the fallback method on the client side shall be implemented.
+In case of multiple artifacts pointing to the same image, the latest will be selected (based on the creation time).
+The client:
+
+1. Fetches the image index/manifest based on the digest or tag.
+2. Iterates over descriptors to check if it's compatibility artifact type.
+3. Pulls the manifest.
+4. Checks if the subject field references the image digest.
```

Comment:
The wording for this is a bit confusing for me. I think we are saying "for each entry in the Index, query for referrers, select the most recent according to the creation annotation if multiple are available, and use the first Index entry with the best matching compatibility spec".

In the above lines, 3 and 4 are unclear if we are talking about the artifact manifest or image manifest, and assuming they are referring to the artifact, what to do with the manifest after pulling it.

Reply:
> In the above lines, 3 and 4 are unclear if we are talking about the artifact manifest or image manifest

I think the description could be enhanced to make it more clear:
1. Fetches the image index based on the digest or tag.
2. Iterates over descriptors in manifests list 
3. Pulls the image manifest according to the descriptor
4. In the fetched image manifest, check to see if it's compatibility artifact type.
5. if it's compatibility artifact type, checks if the subject field references the image digest (which is used in the first step), then the compatibility artifact which points to the image can be found

Reply:
@sudo-bmitch I changed the wording.